### ISSUE TITLE
alpaka::mem::view::copy takes non-const references to views 

### ISSUE THREAD

**Issue by krzikalla** [alpaka-group/alpaka#643]
Follow-up to #642 

Assuming I have a function template giving me a view I cannot write the following:

```
template<typename TDev, typename TElem, typename TExtent>
static auto getViewPlainPtr(TElem* ptr, TDev const& dev, TExtent const & extent)
{
  return mem::view::ViewPlainPtr<TDev, TElem, alpaka::dim::DimInt<1>, size_t>(ptr, dev, extent);
}

mem::view::copy(stream, getViewPlainPtr(...), buffer, size);
```

The problem is, that copy takes the destination view as a non-const reference. However the constness of the memory pointed to cannot be expressed by the constness of the view but must be encoded internally (STL solves this via iterator/const_iterator). A solution could be to handle Elem and const Elem template paramters appropriately.

**Comment by BenjaminW3** [comment]
This is a limitation of the current version. I would prefer to express this via `const Elem` but we have to think through the implications of such a change.

**Comment by bernhardmgruber** [comment]
This issue will be solved by #1743.
